systemvm/debian/opt/cloud/bin/cs/CsAddress.py

```diff
@@ -524,15 +524,16 @@ def post_config_change(self, method):
                     CsHelper.execute("sudo ip route add throw " + self.config.address().dbag['eth1'][0]['network'] + " table " + tableName + " proto static")
 
                 # add 'defaul via gateway' rule in the device specific routing table
-                if "gateway" in self.address and self.address["gateway"] != "None":
+                if "gateway" in self.address and self.address["gateway"] and self.address["gateway"] != "None":
```

Comment:
isn't there a '!= None` missing in this expression? (x != None and x != "None") just paranoia, i guess

Reply:
Why do we check here on "None" as a String? And not None as you should in Python?

Reply:
I think because somewhere the python None is stringyfied and gets in the way, @wido